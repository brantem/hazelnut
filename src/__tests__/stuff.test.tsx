import { act, render, screen, fireEvent, waitFor, within } from '@testing-library/react';
import '@testing-library/jest-dom';

import Stuff from 'pages/stuff';

vi.mock('next/router', () => ({
  useRouter() {
    return {
      pathname: '/',
    };
  },
}));

beforeEach(() => {
  const mockIntersectionObserver = vi.fn();
  mockIntersectionObserver.mockReturnValue({ observe: () => null, unobserve: () => null, disconnect: () => null });
  window.IntersectionObserver = mockIntersectionObserver;
});

test('Stuff', async () => {
  const { container } = render(<Stuff />);

  expect(container).toMatchSnapshot();

  // add group
  {
    expect(screen.queryAllByTestId('group-card')).toHaveLength(0);
    act(() => screen.getByText('Add Group').click());
    expect(screen.getByTestId('save-group-modal')).toBeInTheDocument();
    act(() => {
      fireEvent.change(screen.getByLabelText('Title'), { target: { value: 'Group 1' } });
      screen.getByText('Add').click();
    });
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    expect(screen.queryByTestId('save-group-modal')).not.toBeInTheDocument();
    expect(screen.queryAllByTestId('group-card')).toHaveLength(1);
  }

  // add item
  {
    act(() => screen.getByText('Add Item').click());
    expect(screen.getByTestId('add-group-item-modal')).toBeInTheDocument();
    act(() => {
      fireEvent.change(screen.getByLabelText('Title'), { target: { value: 'Item 1' } });
      screen.getByText('Add').click();
    });
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    expect(screen.queryByTestId('add-group-item-modal')).not.toBeInTheDocument();
    expect(screen.getByTestId('group-card-items-item').textContent).toEqual('Item 1');
  }

  // edit group
  {
    act(() => screen.getByTestId('group-card-settings').click());
    const groupSettingsModal = screen.getByTestId('group-settings-modal');
    expect(groupSettingsModal).toBeInTheDocument();
    act(() => within(groupSettingsModal).getByText('Edit').click());
    expect(screen.getByTestId('save-group-modal')).toBeInTheDocument();
    act(() => {
      fireEvent.change(screen.getByLabelText('Title'), { target: { value: 'Group 1a' } });
      screen.getByTestId('color-picker-option-amber').click();
      screen.getByText('Save').click();
    });
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));
    expect(screen.queryByTestId('group-settings-modal')).not.toBeInTheDocument();
    expect(screen.queryByTestId('save-group-modal')).not.toBeInTheDocument();
  }

  // delete group
  {
    act(() => screen.getByTestId('group-card-settings').click());
    const groupSettingsModal = screen.getByTestId('group-settings-modal');
    expect(groupSettingsModal).toBeInTheDocument();
    act(() => within(groupSettingsModal).getByText('Delete').click());
    act(() => within(groupSettingsModal).getByText('Confirm').click());
    expect(screen.queryByTestId('group-settings-modal')).not.toBeInTheDocument();
    expect(screen.queryAllByTestId('group-card')).toHaveLength(0);
  }

  expect(container).toMatchSnapshot();
});
