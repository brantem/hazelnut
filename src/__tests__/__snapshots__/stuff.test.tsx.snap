// Vitest Snapshot v1

exports[`Stuff 1`] = `
<div>
  <header
    class="flex items-center justify-between py-3 px-4"
  >
    <div
      class="flex items-center text-2xl font-semibold"
    >
      <div
        class="h-6 w-6"
      >
        <svg
          aria-hidden="true"
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.25 12l8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
       
      <h1
        class="ml-3"
      >
        Home
      </h1>
    </div>
    <button
      class="px-3 py-1.5 text-sm rounded-md hover:bg-neutral-100"
    >
      Add Group
    </button>
  </header>
  <main
    class="flex-1"
  >
    <section
      class="space-y-3"
    />
  </main>
  <nav
    class="flex w-full border-t border-t-neutral-200 divide-x divide-neutral-200"
  >
    <a
      aria-current="page"
      class="flex-1 px-4 py-3 font-medium text-center flex items-center justify-center"
      href="/"
    >
      <div
        class="h-6 w-6"
      >
        <svg
          aria-hidden="true"
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.25 12l8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <span
        class="ml-2"
      >
        Home
      </span>
    </a>
    <a
      class="flex-1 px-4 py-3 font-medium text-center flex items-center justify-center"
      href="/routines"
    >
      <div
        class="h-6 w-6"
      >
        <svg
          aria-hidden="true"
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.125 2.25h-4.5c-.621 0-1.125.504-1.125 1.125v17.25c0 .621.504 1.125 1.125 1.125h12.75c.621 0 1.125-.504 1.125-1.125v-9M10.125 2.25h.375a9 9 0 019 9v.375M10.125 2.25A3.375 3.375 0 0113.5 5.625v1.5c0 .621.504 1.125 1.125 1.125h1.5a3.375 3.375 0 013.375 3.375M9 15l2.25 2.25L15 12"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </a>
    <a
      class="flex-1 px-4 py-3 font-medium text-center flex items-center justify-center"
      href="/stuff"
    >
      <div
        class="h-6 w-6"
      >
        <svg
          aria-hidden="true"
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.25 6.375c0 2.278-3.694 4.125-8.25 4.125S3.75 8.653 3.75 6.375m16.5 0c0-2.278-3.694-4.125-8.25-4.125S3.75 4.097 3.75 6.375m16.5 0v11.25c0 2.278-3.694 4.125-8.25 4.125s-8.25-1.847-8.25-4.125V6.375m16.5 0v3.75m-16.5-3.75v3.75m16.5 0v3.75C20.25 16.153 16.556 18 12 18s-8.25-1.847-8.25-4.125v-3.75m16.5 0c0 2.278-3.694 4.125-8.25 4.125s-8.25-1.847-8.25-4.125"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </a>
  </nav>
</div>
`;

exports[`Stuff 2`] = `
<div>
  <header
    class="flex items-center justify-between py-3 px-4"
  >
    <div
      class="flex items-center text-2xl font-semibold"
    >
      <div
        class="h-6 w-6"
      >
        <svg
          aria-hidden="true"
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.25 12l8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
       
      <h1
        class="ml-3"
      >
        Home
      </h1>
    </div>
    <button
      class="px-3 py-1.5 text-sm rounded-md hover:bg-neutral-100"
    >
      Add Group
    </button>
  </header>
  <main
    class="flex-1"
  >
    <section
      class="space-y-3"
    />
  </main>
  <nav
    class="flex w-full border-t border-t-neutral-200 divide-x divide-neutral-200"
  >
    <a
      aria-current="page"
      class="flex-1 px-4 py-3 font-medium text-center flex items-center justify-center"
      href="/"
    >
      <div
        class="h-6 w-6"
      >
        <svg
          aria-hidden="true"
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.25 12l8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <span
        class="ml-2"
      >
        Home
      </span>
    </a>
    <a
      class="flex-1 px-4 py-3 font-medium text-center flex items-center justify-center"
      href="/routines"
    >
      <div
        class="h-6 w-6"
      >
        <svg
          aria-hidden="true"
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.125 2.25h-4.5c-.621 0-1.125.504-1.125 1.125v17.25c0 .621.504 1.125 1.125 1.125h12.75c.621 0 1.125-.504 1.125-1.125v-9M10.125 2.25h.375a9 9 0 019 9v.375M10.125 2.25A3.375 3.375 0 0113.5 5.625v1.5c0 .621.504 1.125 1.125 1.125h1.5a3.375 3.375 0 013.375 3.375M9 15l2.25 2.25L15 12"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </a>
    <a
      class="flex-1 px-4 py-3 font-medium text-center flex items-center justify-center"
      href="/stuff"
    >
      <div
        class="h-6 w-6"
      >
        <svg
          aria-hidden="true"
          class="h-6 w-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.25 6.375c0 2.278-3.694 4.125-8.25 4.125S3.75 8.653 3.75 6.375m16.5 0c0-2.278-3.694-4.125-8.25-4.125S3.75 4.097 3.75 6.375m16.5 0v11.25c0 2.278-3.694 4.125-8.25 4.125s-8.25-1.847-8.25-4.125V6.375m16.5 0v3.75m-16.5-3.75v3.75m16.5 0v3.75C20.25 16.153 16.556 18 12 18s-8.25-1.847-8.25-4.125v-3.75m16.5 0c0 2.278-3.694 4.125-8.25 4.125s-8.25-1.847-8.25-4.125"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </a>
  </nav>
</div>
`;
